### PR TITLE
TextSpan inside LayoutBuilder Rect error

### PR DESCRIPTION
**Pull request by ricnaaru**
## Description

*Fix error when draw TextSpan inside LayoutBuilder with maxLines*

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/rendering/paragraph.dart [hunk 1]
```diff
@@ -510,12 +515,14 @@ class RenderParagraph extends RenderBox {
...
         final SemanticsNode node = SemanticsNode();
         final SemanticsConfiguration configuration = buildSemanticsConfig(current, start);
+        if (configuration == null) break;
         node.updateWith(config: configuration);
         node.rect = currentRect;
         newChildren.add(node);
       }
       final SemanticsNode node = SemanticsNode();
       final SemanticsConfiguration configuration = buildSemanticsConfig(start, end);
+      if (configuration == null) break;
```
Comment (Piinks) [inline packages/flutter/lib/src/rendering/paragraph.dart]: Separate the control structure expression from its statement

File: packages/flutter/lib/src/rendering/paragraph.dart [hunk 2]
```diff
@@ -510,12 +515,14 @@ class RenderParagraph extends RenderBox {
       if (current != start) {
         final SemanticsNode node = SemanticsNode();
         final SemanticsConfiguration configuration = buildSemanticsConfig(current, start);
+        if (configuration == null) break;
```
Comment (Piinks) [inline packages/flutter/lib/src/rendering/paragraph.dart]: Separate the control structure expression from its statement

**Review by Piinks — CHANGES_REQUESTED** [review]
Hi @ricnaaru! Thank you for the contribution! 
It looks like this PR is failing some style related tests. Feel free to check out the [Flutter Style Guide](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo) and see the test output to fix these things.

File: packages/flutter/lib/src/rendering/paragraph.dart [hunk 3]
```diff
@@ -488,6 +488,11 @@ class RenderParagraph extends RenderBox {
         rect = rect.expandToInclude(textBox.toRect());
         currentDirection = textBox.direction;
       }
+      
+      if (rect == null) {
```
Comment (dnfield) [inline packages/flutter/lib/src/rendering/paragraph.dart]: This seems like a bug to me.  @jonahwilliams would know better, but I suspect we should not end up with a null rect here.

**Comment by dnfield** [comment]
Is there an issue open to discuss this problem?

**Comment by ricnaaru** [comment]
I'm sorry i should've report this issue first, as far as i know there's no open issue that discuss this problem.. i will attach reproduce code later today.. so is this pull request supposed to be closed and moved to issue instead for now?

**Comment by ricnaaru** [comment]
Step to reproduce: 

```
class Ex extends StatefulWidget {
  @override
  _ExState createState() => _ExState();
}

class _ExState extends State<Ex> {

  var xxt = TapGestureRecognizer()..onTap = () {
    print("tap tap tap!");
  };
  @override
  Widget build(BuildContext context) {
    List<TextSpan> lts = [TextSpan(style: TextStyle(color: Color(0xff000000), fontSize: 14.0), text: "When Thor's evil brother, Loki (Tom Hiddleston), gains access to the unlimited power of the energy cube called the Tesseract, Nick Fury (Samuel L. Jackson), director of S.H.I.E.L.D., initiates a superhero recruitment effort to defeat the unprecedented threat to Earth. Joining Fury's \"dream team\" are Iron Man (Robert Downey Jr.), Captain America (Chris Evans), the Hulk (Mark Ruffalo), Thor (Chris Hemsworth), the Black Widow (Scarlett Johansson) and Hawkeye (Jeremy Renner)."),
    TextSpan(style: TextStyle(color: Color(0xff2196f3)), recognizer: xxt, text: "https://www.youtube.com/watch?v=hA6hldpSTF8"),];

    return Scaffold(body: RichText(
      text: TextSpan(children: lts),
      maxLines: 3,
    ),);
  }
}
```

My fix may not be the best solution though, i'm sorry to propose it to be a pull request

**Comment by dnfield** [comment]
@jonahwilliams @ricnaaru are we going to carry this forward, or should we just open an issue for it for now and close it?

**Comment by ricnaaru** [comment]
let's just close this pull request and i'll open new issue about it.. I'm sorry that i create a pull request first without an open issue..

**Comment by dnfield** [comment]
No worries!  Thanks for working on this.  Please feel free to tag me and/or @jonahwilliams in the issue.

**Pull request closed without merging**